cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -368,11 +369,18 @@ public void performAction(HttpRequest request, HttpResponder responder,
     }
 
     if (!isValidAction(action)) {
-      responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
```

Comment:
No it does not. Looks like we have some mixed style of error handling in this class:

https://github.com/caskdata/cdap/blob/feature/cdap-2258-better-errors/cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java#L231

vs

https://github.com/caskdata/cdap/blob/feature/cdap-2258-better-errors/cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java#L257